hack/update-codecgen.sh

```diff
@@ -52,7 +98,9 @@ for generated_file in ${generated_files}; do
 	rm -f "${generated_file}"
 done
 
-for generated_file in ${generated_files}; do
+# Generate files in the dependency order.
+for current in ${index[@]}; do
```

Comment:
Added quotes.

Regarding the first comment - personally I don't think it will be simpler - we need to check if a file has dependencies anyway, so this will more-or-less be the same.